In [1]:
import numpy as np
import pandas as pd

In [2]:
dataframe = pd.read_csv("../data/raw/annual-population-growth/annual-population-growth.csv")

### ACTIVIDAD 1:

#### ¿Quién produjo originalmente los datos?<br>
World Population Prospects
#### ¿Qué organización los procesa para su utilización en Our World in Data?<br>
Our World in Data
#### ¿Qué representa la variable que analizarás?<br>
el crecimiento anual de la poblacion en los paises


In [3]:
dataframe.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


In [4]:
dataframe.shape

(38400, 5)

In [5]:
dataframe.columns.tolist()

['Entity',
 'Code',
 'Year',
 'Annual change in population',
 'Annual population change (Projected)']

In [6]:
dataframe.dtypes

Entity                                      str
Code                                        str
Year                                      int64
Annual change in population             float64
Annual population change (Projected)    float64
dtype: object

In [7]:
dataframe.describe()

,Year,Annual change in population,Annual population change (Projected)
count,38400.000000,1.868800e+04,1.971200e+04
mean,2025.500000,2.063019e+06,8.375481e+05
std,43.300872,9.440859e+06,5.816745e+06
min,1951.000000,-3.315935e+06,-2.291035e+07
25%,1988.000000,1.781000e+03,-9.742750e+03
50%,2025.500000,4.689800e+04,9.350000e+01
75%,2063.000000,3.021128e+05,1.411160e+05
max,2100.000000,9.337137e+07,7.206736e+07


In [8]:
dataframe.isnull().sum()

Entity                                      0
Code                                     1200
Year                                        0
Annual change in population             19712
Annual population change (Projected)    18688
dtype: int64

In [9]:
dataframe["Entity"].nunique() 

256

### ACTIVIDAD 2:
#### ¿Cuántas filas y columnas tiene el dataset?
38400 filas y 5 columnas

#### ¿Qué tipo de datos contiene la variable Year?
int64

#### ¿Existen valores faltantes?
si, 1200 de code, 19712 de Annual population change y 18688 en Annual popolation change (projected)

#### ¿Cuántas entidades diferentes contiene el dataset?
256

In [10]:
df_2019 = dataframe[dataframe['Year'] == 2019]
poblacion_2019 = df_2019['Annual change in population']

entidad_min = df_2019.loc[poblacion_2019.idxmin()]["Entity"]
entidad_max = df_2019.loc[poblacion_2019.idxmax()]["Entity"]

print("media =", poblacion_2019.mean())
print("mediana =", poblacion_2019.median())
print("varianza =", poblacion_2019.var())
print("desviacion_std =", poblacion_2019.std())
print("minimo =", poblacion_2019.min(), (entidad_min))
print("maximo =", poblacion_2019.max(), (entidad_max))

media = 2261236.08984375
mediana = 43636.5
varianza = 99503089007422.66
desviacion_std = 9975123.508379364
minimo = -869495.0 Venezuela
maximo = 81390920.0 World


### ACTIVIDAD 3:
#### ¿Cuál es el valor de la media?
Es de 2261236.08984375
#### ¿Cuál es el valor de la mediana?
Es de 43636.5
#### ¿Cuál es el valor de la desviación estándar?
Es de 9975123.508379364
#### ¿Qué entidad presenta el valor mínimo y cuál el máximo?
El valor minimo diria yo que es las personas que mas murieron que en este caso fue en venezuela y supongo que el maximo es las personas que nacieron en ese año

In [84]:
import sys
import notebook

print("Python:", sys.version)
print("Pandas:", pd.__version__)
print("NumPy:", np.__version__)
print("Jupyter:", notebook.__version__)


Python: 3.14.7 (tags/v3.14.7:823f032, Aug  5 2026, 10:51:32) [MSC v.1944 64 bit (AMD64)]
Pandas: 3.0.5
NumPy: 2.5.2
Jupyter: 7.6.2


## Inicio de practica 2


In [12]:
dataframe.head(3)

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN


In [13]:
#Series df_year =dataframe["Year"]

df_year =dataframe[["Year", "Entity"]]
df_year.head()

,Year,Entity
0,1951,Afghanistan
1,1952,Afghanistan
2,1953,Afghanistan
3,1954,Afghanistan
4,1955,Afghanistan


In [14]:
df_extraccion_columnas =dataframe[["Year", "Entity"]]
df_extraccion_columnas.head()

,Year,Entity
0,1951,Afghanistan
1,1952,Afghanistan
2,1953,Afghanistan
3,1954,Afghanistan
4,1955,Afghanistan


In [15]:
df_extraccion_filas =dataframe[dataframe["Year"] == 2010]
df_extraccion_filas.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
59,Afghanistan,AFG,2010,817986.0,NaN
209,Africa (UN),UN_AFR,2010,27438963.0,NaN
359,Albania,ALB,2010,-32756.0,NaN
509,Algeria,DZA,2010,697795.0,NaN
659,American Samoa,ASM,2010,-296.0,NaN


In [ ]:
df_2020_mex = dataframe[(dataframe["Entity"] == "Mexico") & ( dataframe["Year"].between(2003,2020))] [["Entity","Year","Annual change in population"]]
df_2020_mex.head() 


,Entity,Year,Annual change in population
22252,Mexico,2003,1429890.0
22253,Mexico,2004,1415614.0
22254,Mexico,2005,1417373.0
22255,Mexico,2006,1442162.0
22256,Mexico,2007,1520690.0


In [17]:
df_2020_mex.mean(numeric_only=True)

Year                           2.011500e+03
Annual change in population    1.402801e+06
dtype: float64

In [21]:
df_new = dataframe[["Entity"]]
df_new

,Entity
0,Afghanistan
1,Afghanistan
2,Afghanistan
3,Afghanistan
4,Afghanistan
...,...
38395,Zimbabwe
38396,Zimbabwe
38397,Zimbabwe
38398,Zimbabwe


In [37]:
dataframe.loc[:,["Entity","Year"]]

,Entity,Year
0,Afghanistan,1951
1,Afghanistan,1952
2,Afghanistan,1953
3,Afghanistan,1954
4,Afghanistan,1955
...,...,...
38395,Zimbabwe,2096
38396,Zimbabwe,2097
38397,Zimbabwe,2098
38398,Zimbabwe,2099


In [45]:
dataframe.iloc[0:4,0:12]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN


In [47]:
dataframe.loc[10000]

Entity                                  Equatorial Guinea
Code                                                  GNQ
Year                                                 2051
Annual change in population                           NaN
Annual population change (Projected)              46407.0
Name: 10000, dtype: object

In [49]:
dataframe.loc[:,["Entity","Year","Annual change in population"]]

,Entity,Year,Annual change in population
0,Afghanistan,1951,103163.0
1,Afghanistan,1952,108441.0
2,Afghanistan,1953,108919.0
3,Afghanistan,1954,111251.0
4,Afghanistan,1955,119027.0
...,...,...,...
38395,Zimbabwe,2096,NaN
38396,Zimbabwe,2097,NaN
38397,Zimbabwe,2098,NaN
38398,Zimbabwe,2099,NaN


In [50]:
dataframe.loc[10000,["Entity","Year","Annual change in population"]]

Entity                         Equatorial Guinea
Year                                        2051
Annual change in population                  NaN
Name: 10000, dtype: object

In [56]:
dataframe.loc[(dataframe["Entity"] == "Mexico") & ( dataframe["Year"] == 2020)]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
22269,Mexico,MEX,2020,1036075.0,NaN


In [60]:
dataframe.iloc[0:5,]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


In [63]:
dataframe.iloc[:,[0,2,3]]

,Entity,Year,Annual change in population
0,Afghanistan,1951,103163.0
1,Afghanistan,1952,108441.0
2,Afghanistan,1953,108919.0
3,Afghanistan,1954,111251.0
4,Afghanistan,1955,119027.0
...,...,...,...
38395,Zimbabwe,2096,NaN
38396,Zimbabwe,2097,NaN
38397,Zimbabwe,2098,NaN
38398,Zimbabwe,2099,NaN


In [64]:
dataframe.iloc[0:4,0:3]

,Entity,Code,Year
0,Afghanistan,AFG,1951
1,Afghanistan,AFG,1952
2,Afghanistan,AFG,1953
3,Afghanistan,AFG,1954


### Actividad 3 (Ahora correcta)

In [79]:
df_poland = dataframe.loc[(dataframe["Entity"] == "Poland") & (dataframe["Year"] >= 2005) & (dataframe["Year"] <= 2023)]
df_poland_a = dataframe.loc[(dataframe["Entity"] == "Poland") & (dataframe["Year"]  >= 2005) & (dataframe["Year"] <= 2023),"Annual change in population"]
df_poland

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
27654,Poland,POL,2005,-43191.0,NaN
27655,Poland,POL,2006,-53208.0,NaN
27656,Poland,POL,2007,-50749.0,NaN
27657,Poland,POL,2008,-24518.0,NaN
27658,Poland,POL,2009,6251.0,NaN
27659,Poland,POL,2010,93203.0,NaN
27660,Poland,POL,2011,113288.0,NaN
27661,Poland,POL,2012,39223.0,NaN
27662,Poland,POL,2013,11400.0,NaN
27663,Poland,POL,2014,5846.0,NaN


In [82]:
print("La Media de poland es:", df_poland_a.mean())
print("La Mediana de poland es:", df_poland_a.median())
print("La Varianza de poland es:", df_poland_a.var())
print("La Desviacion estandar de poland es:", df_poland_a.std())
print("El Minimo de poland es:", df_poland_a.min())
print("EL Maximo de poland es:", df_poland_a.max())

La Media de poland es: 31330.473684210527
La Mediana de poland es: -3054.0
La Varianza de poland es: 16367786520.26316
La Desviacion estandar de poland es: 127936.65041833463
El Minimo de poland es: -130702.0
EL Maximo de poland es: 377113.0
